router.go

```diff
@@ -34,6 +37,15 @@ func GetConn(ctx context.Context) net.Conn {
 	return nil
 }
 
+// GetRouter gets the `*Router` handling API requests
+func GetRouter(ctx context.Context) *Router {
```

Comment:
Can this be used to greatly simplify the code in #80? The context would already have the router injected and there's no need to pass it around everywhere right? 🤔

Reply:
Possibly.  Note that several tests don't have a router (in which case I send in `nil`).  But since the `GetRouter` function returns `nil` when it can't find a `Router` in the context, then it should actually be pretty equivalent.  I agree, it would simplify the code.  The issue here was that I wanted to provide you with independent PRs in case you decided to reject any.  So that's why I don't leverage this in #80.  But if you want me to refactor it once this PR is merged, I can certainly do that.